executor/src/witgen/jit/effect.rs

```diff
@@ -144,6 +149,12 @@ pub fn format_code<T: FieldElement>(effects: &[Effect<T, Variable>]) -> String {
             Effect::RangeConstraint(..) => {
                 panic!("Range constraints should not be part of the code.")
             }
+            Effect::ProverFunctionCall(v, index, args) => {
+                format!(
+                    "{v} = prover_function_{index}([{}]);",
```

Comment:
so the generated code for these takes in an array of arguments? just to know, why is that?

Reply:
In the end, it calls a PIL function that takes an array. It would be better to take a fixed-size slice or something, but then we can't do it in a generic way. In the end, we do pass a slice and the code that destructures the array again is in closures that are local to a function. If we are lucky, the rust optimizer can disentangle everything. I'll take a look at the assembly. Even if not, I think it's OK, performance-wise.